src/Playwright/Core/BrowserContext.cs

```diff
@@ -251,6 +251,9 @@ public async Task<IPage> NewPageAsync()
         public Task RouteAsync(string url, Action<IRoute> handler, BrowserContextRouteOptions options = default)
             => RouteAsync(new Regex(CombineUrlWithBase(url).GlobToRegex()), null, handler, options);
 
+        public Task RouteAsync(string url, Task<IRoute> handler, BrowserContextRouteOptions options = null)
```

Comment:
Is this correct?

```suggestion
        public Task RouteAsync(string url, Func<IRoute, Task> handler, BrowserContextRouteOptions options = null)
```


Should there also be another overload:

```c#
public Task RouteAsync(Func<string, bool> url, Func<IRoute, Task> handler, BrowserContextRouteOptions options = default)
```

Reply:
I think more than "another" that should be the right one.

Reply:
thanks